nmigen_boards/alchitry_au.py

```diff
@@ -0,0 +1,86 @@
+import os
+import subprocess
+
+from nmigen.build import *
+from nmigen.vendor.xilinx_7series import *
+from .resources import *
+
+
+__all__ = ["AlchitryAuPlatform"]
+
+def find_loader():
+    import shutil # Delay this until we're actually trying to program
+    loader_prgm = os.environ.get("ALCHITRY_LOADER", shutil.which("loader"))
+    if loader_prgm is None:
+        raise EnvironmentError("Could not find Alchrity Loader. Place "
+            "it directly in PATH or specify path explicitly via the "
+            "ALCHITRY_LOADER environment variable")
+    bridge_bin = os.environ.get("ALCHITRY_BRIDGE_BIN", os.path.join(os.path.dirname(loader_prgm), "au_loader.bin"))
+    return (loader_prgm, bridge_bin)
+
+
+class AlchitryAuPlatform(Xilinx7SeriesPlatform):
+    device      = "XC7A35T" # Artix 7 33K LEs
+    package     = "FTG256"
+    speed       = "1"
+    default_clk = "clk100"
+    resources   = [
+        Resource("clk100", 0, Pins("N14", dir="i"),
+                 Clock(10e7), Attrs(IOSTANDARD="LVCMOS33")),
+
+        # On-Board LED Array
+        *LEDResources(
+            pins="K13 K12 L14 L13 M16 M14 M12 N16",
+            attrs=Attrs(IOSTANDARD="LVCMOS33")),
+
+        Resource("usb", 0,
+            Subsignal("usb_tx", Pins("P16", dir="o")),
+            Subsignal("usb_rx", Pins("P15", dir="i")),
+            Attrs(IOSTANDARD="LVCMOS33")
+        ),
+
+        # TODO: This is untested
+        Resource("ddr3", 0,
+            Subsignal("rst",     PinsN("D13", dir="o")),
+            Subsignal("clk",     DiffPairs("G14", "F14", dir="o"), Attrs(IOSTANDARD="LVDS")),
+            Subsignal("clk_en",  Pins("D15", dir="o")),
+            Subsignal("cs",      PinsN("D16", dir="o")),
+            Subsignal("we",      PinsN("E11", dir="o")),
+            Subsignal("ras",     PinsN("D11", dir="o")),
+            Subsignal("cas",     PinsN("D14", dir="o")),
+            Subsignal("a",       Pins("F12 G16 G15 E16 H11 G12 H16 H12 H16 H13 E12 H14 F13 J15", dir="o")),
+            Subsignal("ba",      Pins("E13 F15 E15", dir="o")),
+            Subsignal("dqs",     DiffPairs("B15 A15", "B9 A10", dir="io"), Attrs(IOSTANDARD="LVDS")),
+            Subsignal("dq",      Pins("A13 B16 B14 C11 C13 C16 C12 C14 D8 B11 C8 B10 A12 A8 B12 A9",
+                                      dir="io")),
+            Subsignal("dm",      Pins("A14 C9", dir="o")),
+            Subsignal("odt",     Pins("G11", dir="o")),
+            Attrs(IOSTANDARD="LVCMOS15")
+        )
+
+    ]
+
+    connectors  = [
+        Connector("bank", 0, "T8  T7  T5  R5  R8  P8  L2  L3  J1  K1  H1  H2  G1  G2  K5  E6"
```

Comment:
This is unfortunately wrong because you ended up defining a pin "E6T10" (there's no space at the end). I'll fix this in this PR, and I'll try to fix this properly in nmigen.build (probably allow using multiple string arguments).